### DISCUSSION TITLE
Code tweak: new "mode" prop.

### DISCUSSION THREAD

**Discussion by HenriqueSPin** [laravel/nova-issues#4738] [Ideas & Feature Requests] [answered]
In FormField.js, the prop "mode" is being declared instead of mapped via mapProps.

It is now:

```ES6
  props: {
    ...mapProps([
      'shownViaNewRelationModal',
      'field',
      'viaResource',
      'viaResourceId',
      'viaRelationship',
      'resourceName',
      'showHelpText',
    ]),
    formUniqueId: {
      type: String,
    },
    mode: {
      type: String,
      default: 'form',
      validator: v => ['form', 'modal'].includes(v),
    },
  },
```

It should be:
```ES6
  props: {
    ...mapProps([
      'shownViaNewRelationModal',
      'field',
      'viaResource',
      'viaResourceId',
      'viaRelationship',
      'resourceName',
      'showHelpText',
      'mode'
    ]),
    formUniqueId: {
      type: String,
    },
  },
```

Also... I believe the DependentFormField.js should have this prop.

**Comment by crynobone** [comment, marked as answer]
Changes to `FormField` has been made, changes to `DependentFormField` is not required since it doesn't use it.

**Reply by HenriqueSPin** [reply]
Tks @crynobone. Don’t you think that this could be useful for custom fields using DependentFormField? It would be nice to know if a dependent field is in a modal or not.

**Reply by crynobone** [reply]
it's a mixin, the custom field still need to define the props if the component need to use it. Also, `mapTypes` is available from `laravel-nova` alias import.
